packages/native/src/useScrollToTop.tsx

```diff
@@ -82,14 +83,16 @@ export default function useScrollToTop(
           const scrollable = getScrollableNode(ref);
 
           if (isFocused && isFirst && scrollable && !e.defaultPrevented) {
+            const contentOffsetY = scrollable.props.contentOffset?.y || 0;
```

Comment:
Thanks for the PR! I think this isn't the right solution since this relies an implementation detail. Since the scrollables are usually class components, it's possible to access their props via `ref.props` (since `ref` is the instance of the class), but say if someone refactprs them to a function component in future and uses `useImperativeHandle` to set the ref instead, this code will break. imo the approach I suggested in #6796 will be the most flexible and future proof.

Reply:
Thanks for you fast answer and review! You're right it's not ok to rely on `ref.props`.
I tried this approach because your is more flexible indeed, but need to write much implementation detail already written in the hook implementation (like the `getScrollableNode` function) and it also need to knowing the implementation that `scrollToTop` is prioritized and will not change in the future.

For example, for a FlatList wrapped in Animated:
```js
useScrollToTop(useRef({
  scrollToTop: () => scrollRef.current?.getNode().scrollToOffset({ offset: contentOffset.y }),
}));
```
(and i'm not a fan of the polymorphic `ref | () => void` of the simpler example, but it's subjective).

I think the only thing we want to change in the `useScrollToTop` hook is the hardcoded `0` that assume doing a `scrollToTop` and so that why I understand @christianjuth proposal to give it as a param.

But more I think about it, more I think it's not the job of react-navigation and a `scrollToTop`/`scrollToStart` should exist for `<ScrollView>` directly in react-native. The function will take into consideration the `contentOffset` prop or, I don't know the implementation detail, called the native fonction 'scrollToTop' equivalent. The 'scrollToTop' work out of the box with `contentOffset` and `contentInset` if we touch the StatusBar so I think something like this already exist in the native part or in the react-native implementation.

Maybe you're already know something about it on the react-native part ? That's why `scrollToTop` is prioritized ?

For now I'm closing this issue.

Reply:
Ok, I found a new use case where we need more flexibility so your design is better.
If you set an `initialScrollIndex` you want to use `scrollToIndex()` for `useScrollToTop()`.